### PR TITLE
Used cached doc position in getLayoutRect.

### PR DESCRIPTION
**Pull request by cramforce**
This showed up benchmarking real world docs.

### CODE REVIEW

**Comment by mkhatib** [comment]
Does this address #2424?

**Comment by cramforce** [comment]
@mkhatib Nope that is separate, but good question!

File: src/service/viewport-impl.js [hunk 1]
```diff
@@ -686,9 +687,9 @@ export class ViewportBindingNatural_ {
   }
 
   /** @override */
-  getLayoutRect(el) {
-    const scrollTop = this.getScrollTop();
-    const scrollLeft = this.getScrollLeft();
+  getLayoutRect(el, opt_scrollLeft, opt_scrollTop) {
+    const scrollTop = opt_scrollTop || this.getScrollTop();
```
Comment (dvoytenko) [inline src/service/viewport-impl.js]: Do you want to do this on `0`?

File: src/service/viewport-impl.js [hunk 2]
```diff
@@ -686,9 +687,9 @@ export class ViewportBindingNatural_ {
   }
 
   /** @override */
-  getLayoutRect(el) {
-    const scrollTop = this.getScrollTop();
-    const scrollLeft = this.getScrollLeft();
+  getLayoutRect(el, opt_scrollLeft, opt_scrollTop) {
```
Comment (dvoytenko) [inline src/service/viewport-impl.js]: Let's add a comment to `onScroll` to explain that cached value must be updated or invalidated.

Comment (cramforce) [inline src/service/viewport-impl.js]: Done

File: src/service/viewport-impl.js (hunk 1 above)
Comment (cramforce) [inline src/service/viewport-impl.js]: No! Fixed.

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**